File: Year-2023/LeetCode-July-Challenge/PeakIndexinaMountainArray.js
var peakIndexInMountainArray = function (arr) {
    let low = 0, high = arr.length - 1;
    while (low < high) {
        let mid = low + Math.trunc((high - low) / 2);
        if (arr[mid] < arr[mid + 1]) low = mid + 1;
        else high = mid;
    }
    return low;
};

let arr = [0,1,0];
console.log(peakIndexInMountainArray(arr));

File: Year-2023/LeetCode-July-Challenge/PeakIndexinaMountainArray.java
class Solution {
    public int peakIndexInMountainArray(int[] arr) {
        int low = 0, high = arr.length - 1;
        while(low < high){
            int mid = low + (high - low)/2;
            if(arr[mid] < arr[mid + 1]) low = mid + 1;
            else high = mid;
        }
        return low;
    }
}
public class PeakIndexinaMountainArray{
    public static void main(String[] args) {
        int[] arr = {0,1,0};
        System.out.println(new Solution().peakIndexInMountainArray(arr));
    }
}